src/cascadia/TerminalApp/WslDistroGenerator.cpp

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+
+#include "WslDistroGenerator.h"
+#include "LegacyProfileGeneratorNamespaces.h"
+
+#include "../../types/inc/utils.hpp"
+#include "../../inc/DefaultSettings.h"
+#include "Utils.h"
+#include <io.h>
+#include <fcntl.h>
+#include "DefaultProfileUtils.h"
+
+using namespace ::TerminalApp;
+
+// Legacy GUIDs:
+//   - Debian       58ad8b0c-3ef8-5f4d-bc6f-13e4c00f2530
+//   - Ubuntu       2c4de342-38b7-51cf-b940-2309a097f518
+//   - Alpine       1777cdf0-b2c4-5a63-a204-eb60f349ea7c
+//   - Ubuntu-18.04 c6eaf9f4-32a7-5fdc-b5cf-066e8a4b1e40
+
+std::wstring_view WslDistroGenerator::GetNamespace()
+{
+    return WslGeneratorNamespace;
+}
+
+// Method Description:
+// -  Enumerates all the installed WSL distros to create profiles for them.
+// Arguments:
+// - <none>
+// Return Value:
+// - a vector with all distros for all the installed WSL distros
+std::vector<TerminalApp::Profile> WslDistroGenerator::GenerateProfiles()
+{
+    std::vector<TerminalApp::Profile> profiles;
+
+    wil::unique_handle readPipe;
+    wil::unique_handle writePipe;
+    SECURITY_ATTRIBUTES sa{ sizeof(sa), nullptr, true };
+    THROW_IF_WIN32_BOOL_FALSE(CreatePipe(&readPipe, &writePipe, &sa, 0));
+    STARTUPINFO si;

```

Comment:
@zadjii-msft I saw you removed the {} for STARTUPINFO, in this case you should initialize it so all fields are initialized.  It is a struct and not an object.

STARTUPINFO si{};
// or
STARTUPINFO si = {0};

I'm not sure what the team's preference is.